test/Microsoft.Azure.Devices.Edge.Test.Common/linux/PackageManagement.cs

```diff
@@ -0,0 +1,187 @@
+            this.isInitialized = false;
+        }
+
+        async Task CheckInitializationAsync()
+        {
+            if (!this.isInitialized)
+            {
+                var tokenSource = new CancellationTokenSource();
+                string[] platformInfo = await Process.RunAsync("lsb_release", "-sir", tokenSource.Token);
+                this.InitializeParameters(platformInfo);
+                this.isInitialized = true;
+            }
+        }
+
+        void InitializeParameters(string[] platformInfo)
+        {
+            if (platformInfo.Length == 1)
+            {
+                platformInfo = platformInfo[0].Split(' ');
+            }
+
+            string os = platformInfo[0].Trim();
+            string version = platformInfo[1].Trim();
+
+            switch (os)
+            {
+                case "Ubuntu":
+                    this.os = "ubuntu";
+                    this.version = version;
+                    this.packageTool = "apt-get";
+                    this.packageExtension = SupportedPackageExtension.Deb;
+                    this.forceInstallConfigCmd = "dpkg --force-confnew -i";
+                    this.installCmd = "install -f";
+                    this.uninstallCmd = "purge --yes";
+                    this.iotedgeServices = "iotedge.mgmt.socket iotedge.socket iotedge.service";
+                    break;
+                case "Raspbian":
+                    this.os = "debian";
+                    this.version = "stretch";
+                    this.packageTool = "apt-get";
+                    this.packageExtension = SupportedPackageExtension.Deb;
+                    this.forceInstallConfigCmd = "dpkg --force-confnew -i";
+                    this.installCmd = "install -f";
+                    this.uninstallCmd = "purge --yes";
+                    this.iotedgeServices = "iotedge.mgmt.socket iotedge.socket iotedge.service";
+                    break;
+                case "CentOS":
+                    this.os = os.ToLower();
+                    this.version = version.Split('.')[0];
+                    this.packageTool = "yum";
+                    this.packageExtension = SupportedPackageExtension.Rpm;
+                    this.forceInstallConfigCmd = "rpm -iv --replacepkgs";
+                    this.installCmd = "install -y";
+                    this.uninstallCmd = "remove -y --remove-leaves";
+                    this.iotedgeServices = "iotedge.service";
+
+                    if (this.version != "7")
+                    {
+                        throw new NotImplementedException($"Don't know how to install daemon on operating system '{this.os} {this.version}'");
+                    }
+
+                    break;
+                default:
+                    throw new NotImplementedException($"Don't know how to install daemon on operating system '{os}'");
+            }
+        }
+
+        public async Task<string[]> GetInstallCommandsFromLocalAsync(string path)
```

Comment:
There are four public methods in this class. Two return an array of command strings that the caller must pass to `Process.RunAsync`, and the other two run the commands directly. I think you should choose one approach or the other for all four methods.

Reply:
yup. consolidating